### PR TITLE
Dreamchecker fixes

### PR DESCRIPTION
**Pull request by adacovsk**
[bugfix]
Getting compiler warnings. Did not test, please review. I can at least compile now.

### CODE REVIEW

File: code/game/objects/items/weapons/cosmetics.dm [hunk 1]
```diff
@@ -525,7 +525,7 @@
 				to_chat(H, "<span class='notice'>You don't see anything.</span>")
 				return
 		if(user.hallucinating())
-			switch(rand(1,100))
```
Comment (MadmanMartian) [inline code/game/objects/items/weapons/cosmetics.dm]: What the

File: code/game/objects/items/weapons/power_cells.dm [hunk 2]
```diff
@@ -270,7 +270,7 @@
 /obj/item/weapon/cell/rad/emp_act(severity)
 	..()
 	if(maxcharge > 0)
-		switch(rand(3))
```
Comment (MadmanMartian) [inline code/game/objects/items/weapons/power_cells.dm:273]: What the

File: code/game/objects/structures/mirror.dm [hunk 3]
```diff
@@ -53,7 +53,7 @@
 	if(!can_use(user, target))
 		return
 	if(user.hallucinating())
-		switch(rand(1,100))
```
Comment (MadmanMartian) [inline code/game/objects/structures/mirror.dm]: What the

**Comment by Exxion** [comment]
The ones that change `||` to `,` will probably fix a lot of actual bugs. The ones that change the `rand()`s, though, will significantly change intended behavior. I can't check because I still haven't gotten BYOND set up on my new computer, but I'm guessing the linter is complaining about possible values outside the checked range? I'd recommend keeping the same values but checking for them, either with another range or with `default`.
Also, I think you mean dreamchecker.
Also, I should update the CI.

**Comment by adacovsk** [comment]
> The ones that change `||` to `,` will probably fix a lot of actual bugs. The ones that change the `rand()`s, though, will significantly change intended behavior. I can't check because I still haven't gotten BYOND set up on my new computer, but I'm guessing the linter is complaining about possible values outside the checked range? I'd recommend keeping the same values but checking for them, either with another range or with `default`. Also, I think you mean dreamchecker. Also, I should update the CI.

you're right

File: code/modules/power/treadmill.dm [hunk 4]
```diff
@@ -85,6 +85,8 @@
 						to_chat(runner,"<span class='warning'>Your legs really hurt!</span>")
 						runner.apply_damage(5, BRUTE, LIMB_LEFT_LEG)
 						runner.apply_damage(5, BRUTE, LIMB_RIGHT_LEG)
+					else
```
Comment (MadmanMartian) [inline code/modules/power/treadmill.dm:88]: What the

File: code/game/objects/items/weapons/cosmetics.dm [hunk 5]
```diff
@@ -539,6 +539,8 @@
 				if(41 to 50)
 					to_chat(H, "<span class='notice'>You don't see anything.</span>")
 					return
+				else
+					return
```
Comment (Exxion) [inline code/game/objects/items/weapons/cosmetics.dm]: Not equivalent. Prevents the next line from being called.

**Comment by DamianX** [comment]
`switch(rand(1,100))` looks silly. Maybe in the future we should use something like pickweight() instead.

**Pull request merged**